src/main/java/com/direwolf20/mininggadgets/common/items/MiningGadget.java

```diff
@@ -97,14 +99,17 @@ public void addInformation(ItemStack stack, @Nullable World world, List<ITextCom
         super.addInformation(stack, world, tooltip, flag);
 
         List<Upgrade> upgrades = UpgradeTools.getUpgrades(stack);
-        if(!Screen.hasShiftDown())
+        Minecraft mc = Minecraft.getInstance();
+
+        if(!InputMappings.isKeyDown(mc.mainWindow.getHandle(), mc.gameSettings.keyBindSneak.getKey().getKeyCode()))
         {
-            // Todo: check if we can get the minecraft instance here for Minecraft.getInstance().gameSettings.KeySneaking
-            tooltip.add(new StringTextComponent("Hold shift to show upgrades").applyTextStyle(TextFormatting.GRAY));
+            tooltip.add(new TranslationTextComponent("mininggadgets.tooltip.item.show_upgrades",
+                    mc.gameSettings.keyBindSneak.getLocalizedName().toLowerCase())
```

Comment:
Did you start a dedicated server (runServer) to confirm that this tooltip does not cause sided issues. I forget which side tooltips tent do be on.

Reply:
Should be strictly client since thats where the langauge file is determined. Did a println and it only printed on client thread so it seems it'd be fine